### PR TITLE
WINC-648: [wmco] Remove HNS networks in BYOH deconfiguration

### PR DESCRIPTION
**Pull request by mansikulkarni96**
The hybrid-overlay process creates HNS networks when it runs on a node. This PR removes the HNS networks created by
previous configuration on the node and adds a test case in the deletion test suite to test the removal.

### CODE REVIEW

**Review by saifshaikh48 — COMMENTED** [review]
Thanks for working on this @mansikulkarni96.

**Comment by sebsoto** [comment]
I'd suggest adding https://github.com/openshift/windows-machine-config-operator/pull/555/commits/3985691975f7645e2d5cde6a0819a9706376977d to this PR, as it fits into this PR well, and it doesnt look like the OVN fix will make it into 3.1.0.

File: test/e2e/delete_test.go
Comment (aravindhp) [inline test/e2e/delete_test.go:95]: I am noticing that `powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command` is being repeated in the test code and warrants a constant.

File: test/e2e/reconfigure_test.go
Comment (aravindhp) [inline test/e2e/reconfigure_test.go:66]: Please use a descriptive variable whenever possible. `windowsInstances` will make more sense here.

File: test/e2e/reconfigure_test.go
Comment (aravindhp) [inline test/e2e/reconfigure_test.go:79]: Shouldn't you be patching?

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @mansikulkarni96. Please address my comments.

File: pkg/windows/windows.go
Comment (jrvaldes) [inline pkg/windows/windows.go:800]: Is there a need to remove both networks (`net0` and `net1`) at once?

**Review by jrvaldes — CHANGES_REQUESTED** [review]
Thanks @mansikulkarni96 for your contribution, comments added PTAL.

File: test/e2e/delete_test.go
Comment (sebsoto) [inline test/e2e/delete_test.go:101]: You should also check that the HNS endpoint created by WMCO is removed as well

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: pkg/windows/windows.go
Comment (aravindhp) [reply pkg/windows/windows.go:252]: @mansikulkarni96 I am wondering if the issue you are running into is that when you delete the HNS networks, it causes the networks to be reconfigured thereby losing connectivity. This could be manifesting itself by taking too long to return. Please see [comment](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/windows/windows.go#L375).

Comment (mansikulkarni96) [reply pkg/windows/windows.go:252]: I understand, that is exactly why this hack is needed. Are you suggesting a workaround for that, not sure I got that part?

Comment (aravindhp) [reply pkg/windows/windows.go:252]: Point I am making is that the hack is too generic.

File: pkg/windows/windows.go
Comment (aravindhp) [reply pkg/windows/windows.go:77]: >It is returned when the HNS network deletion command takes too long to return.

To further clarify, there is potential for other commands to take too long to return and not just the cmd to delete HNS networks.

File: pkg/windows/windows.go
Comment (aravindhp) [reply pkg/windows/windows.go:252]: IOW wouldn't we want to know a command other than deleting the HNS networks took too long to run? By suppressing the error message across the board, you are hiding that information.

File: test/e2e/reconfigure_test.go
Comment (aravindhp) [reply test/e2e/reconfigure_test.go:18]: Just ensure that we are not overloading one provider with more tests.

File: test/e2e/reconfigure_test.go
Comment (aravindhp) [inline test/e2e/reconfigure_test.go:23]: Shouldn't this be vSphere given BYOH is mainly going to be used there?

File: pkg/windows/windows.go
Comment (aravindhp) [inline pkg/windows/windows.go:799]: Please add a comment here mentioning the VIP created by WMCO also gets deleted when deleting the HNS networks.

**Review by aravindhp — APPROVED** [review]
Thanks for working on this, @mansikulkarni96 

/approve

**Comment by saifshaikh48** [comment]
/hold

Until the kubelet bump #581 merges.

**Comment by aravindhp** [comment]
@mansikulkarni96 when hybrid-overlay creates the HNS networks, we [reinitialize](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/windows/windows.go#L362) the SSH client before connecting again. Don't we need to do the same here? I think we are not running into issues because we don't execute any commands after we remove the HNS networks.

**Comment by mansikulkarni96** [comment]
@aravindhp that is correct, removal of networks is the last step in the deconfiguration of node. Now that you mention, I think I should add a comment about it.

**Pull request merged**
